content/docs/user-guide/remote-storage.md

```diff
@@ -0,0 +1,200 @@
+  though the file appears in two different versions of the project.
+- You copy an image dataset so you can add new data to it, but want to keep the
+  original dataset unchanged. The new version of the image dataset will contain
+  an entire copy of the entire original dataset.
+- You copy a large file, but don't change the copy, so you have two files with
+  different names and other metadata but the exact same content.
+
+The way that DVC determines if two different files are duplicates comes from a
+computer science idea called
+[content-addressable storage](/doc/user-guide/project-structure/internal-files).
+Basically, a unique hash of the file's _content_ (ignoring file metadata) is
+calculated. For every pair of files with different content, therefore, the
+unique hashes should be different; but if two files have the same content (even
+though they may differ in metadata such as file name and creation time), then
+the hashes will be the same. DVC renames files with their hash and stores the
+real names elsewhere. In this way, it can track files that change, track files
+that don't change, and determine which (apparently different) files are
+identical. DVC does a similar trick for directories, ensuring that directory
+changes (e.g. adding new files) are also detectable and trackable.
+
+💡 The specific type of hash used by DVC currently is the
+[MD5 hash](https://en.wikipedia.org/wiki/MD5).
+
+Let's look at a simple example: the file `a.dat`. After adding the file to the
+project with `dvc add a.dat`, a new file `a.dat.dvc` is created in the project
+folder. The new file contains the hash of `a.dat`, which can be looked at by
+examining the file:
+
+```dvc
+$ cat a.dat.dvc
+outs:
+- md5: bba40b7807c80d3f44787b9c6a4aabee
+  size: 1047565
+  path: a.dat
+```
+
+The original `a.dat` has now been renamed to `bb\a40b7807c80d3f44787b9c6a4aabee`
+and moved to the local cache. The version of `a.dat` that we now see in the
+project directory is now a
+[file link](doc/user-guide/large-dataset-optimization#file-link-types-for-the-dvc-cache)
+that refers to the hash-named version of the file.
+
+💡 Note that DVC supports file links that are
+[reflinks](https://blog.ram.rachum.com/post/620335081764077568/symlinks-and-hardlinks-move-over-make-room-for)
+(the default, if possible), hard links, soft links, and basic copies. Please
+check your operating systems capabilities and configure DVC to use the
+appropriate file link type using `dvc config` if needed.
+
+When the project is committed and pushed, the version of `a.dat` in the cache
+will be copied to remote storage. If `a.dat` changes later and is committed
+again, then a new hash for the file will be computed, a new version of the file
+will be created in the cache, and the file link in the project directory is
+updated.
+
+The next brief example shows a directory `myDir` tracked by DVC containing two
+files `a` and `b`:
+
+![Local cache and remote storage](/img/cache_structure.png)
```

Comment:
This 2nd diagram is confusing though. But again, for now we can leave out all these details about the caching mechanism and storage optimization, which are not specifically about remote storage.